.github/workflows/ignore-e2e.yml

```diff
@@ -0,0 +1,26 @@
+# This workflow is added in order for the required e2e/build job to run successfully even
+# when the PR targets non-code files (e.g. docs, LICENSE, etc). This is done because of the
+# way Github required checks work, reference:
+#
+# https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/troubleshooting-required-status-checks#handling-skipped-but-required-checks
+name: Tests / E2E
+on:
```

Comment:
should we run this on forks?

Reply:
Sorry, my comment was asking if this should be run on forks for code changes, as it seems fork pr's currently hang on required E2E's

Reply:
that's slightly tricker, if we want to make this run on code changes for forks it would run on code changes on regular PRs as well

Reply:
Maybe we can make the existing E2E task `exit 0` if it's a fork and pass?

Reply:
there's some missing context since Colins last comment, seems the hanging was caused, partly, by `e2e-fork.yml` and `e2e,yml` having the same job name for running the tests, `e2e`. A first step would be going with the renaming #3827 and adding the new job names in the branch protection rules.

I think the workflows to be added in this PR are a different issue that's unrelated. These just make sure that the job will be skipped correctly on non-code prs due to the way github reports a job status if it is skipped with a `path` filter on `on`.

Reply:
hopefully, it definitely appears to be the first right step. After it gets merged we can trigger some runs/add it to branch protection and see how it plays.